README.md
Calculator of probabilities.

https://metsuryu.github.io/probabilityCalculator/

Can add named variables with a min, max, and weight values, then they get averaged out by weight, and the resulting range is shown.

I added a preset for my p(doom from AI) calculation, where I calculate the probabilities for AI going catastrophically wrong.

Values in this readme might be outdated, see .js for latest values, this is just plain text to roughly explain the concept.

# Assumptions
- Orthogonality is true = 100-ε%
- Moral realism is false = 100-ε%
- Probability that misalignment is impossible = ε%

# Scenarios
  * Technical [Determines whether alignment is solved]
  - Instrumental convergence applies to AGI = 80-95% Weight:1

  - AGI not achieved with robustly alignable methods = 55-80% Weight:1
  - Interp doesn't hold for stronger models = 60 - 75% Weight:0.9
  - AGI can be deceptively aligned = 5-80% (5% if achieved with robust methods and Interp holds for stronger models, up to 80% if using something else, or interp. becomes harder). Weight:1

  - We don't figure out how to have it prioritize corrigibility = 50-55% Weight:0.6
  - We can't use corrigibility to prevent irreversible catastrophe before it occurs (might not show signs of misalignemnt until it knows we won't be able to correct it) = 90% if deceptive alignment is possible, 15% if not. Weight:0.45
  - Unforeseen alignment issues = 15% Weight:0.6

  * Societal [Does not determine, but influences likelihood of solving alignment (S), and applying it(A)]
  - (S:1) A capability pause doesn't happen or isn't properly enforced = 98% (excludes hardware pause = 99%)
  - (S:1 && A:1) We don't form an international collaboration to work on alignment = 85-90%
  - (S:0.1 && A:0.6) Other coordination issues or non-technical mistakes = 65%

  S is applied to "Alignment not solved" probability after weighted calculation, A is applied to "Alignment solution is not applied" and "Alignment solved and applied, but the AGI is misused".

# Outcomes
  - Alignment not solved to satisfactory degree before we get AGI = 53.3% - 78.5%
  - Alignment solved but not applied properly, or applied, but the AGI is misused = 3.0% - 8.7%
  - Alignment solved and applied = 24.6% - 55.4%
  - Uncertainty = 30%


CHANGELOG.md
# 2023/12/05

- META, IBM, and more form Open Source AI alliance: https://www.reddit.com/r/singularity/comments/18b9xlh/meta_ibm_and_over_50_other_companies_and/
Updating several variables, see diff.


# 2023/12/04

- U.S. Gov warns NVIDIA: https://www.reddit.com/r/technology/comments/18ajefm/us_issues_warning_to_nvidia_urging_to_stop/
Updating noPause, noCollab, otherS, and otherA


probabilityCalculator.js
function getAvg(weighedParamsArray, mode = 'min') {
  let total = 0;
  let allWeights = 0;

  weighedParamsArray.forEach(weighedParam => {
    allWeights += weighedParam.weight;
    if (mode === 'min') {
      total += (weighedParam.min * weighedParam.weight);
    } else{
      total += (weighedParam.max * weighedParam.weight);
    }
  });

  return total / allWeights;
}

function calculateMisalignmentLikelihood(mode = 'min') {
  const orthogonalityTrue = 100 - Number.EPSILON;
  const moralRealismTrue = 100 - Number.EPSILON;
  const misalignmentImpossible = 100 - ((orthogonalityTrue + moralRealismTrue) / 2);

  const instrumentalConvergence = {
    min: 24,
    max: 85,
    weight: 1
  };

  const AGINotAchievedWithRobustMethods = {
    min: 34,
    max: 80,
    weight: 0.8
  };

  const interpFails = {
    min: 24,
    max: 75,
    weight: 0.6
  };

  const deceptiveAlignment = {
    min: 5,
    max: 75,
    weight: 0.8
  };

  const corrigibility = {
    min: 19,
    max: 55,
    weight: 0.8
  };

  const corrigibilityUseless = {
    min: 15,
    max: 90,
    weight: 0.45
  };

  const unforeseenConsequences = {
    min: 14,
    max: 20,
    weight: 0.5
  };

  return getAvg(
    [
      instrumentalConvergence,
      AGINotAchievedWithRobustMethods,
      interpFails,
      deceptiveAlignment,
      corrigibility,
      corrigibilityUseless,
      unforeseenConsequences
    ], mode) - misalignmentImpossible;
}

function calculateSocietalWeights(mode = 'min') {
  const noPause = {
    min: 89,
    max: 98,
    weight: 1
  };

  const noCollab = {
    min: 85,
    max: 90,
    weight: 1
  };

  /* Influences probability of solving alignment */
  const otherSolve = {
    min: 38,
    max: 63,
    weight: 0.1
  };

  /* Influences probability of applying alignment */
  const otherApply = {
    min: 40,
    max: 65,
    weight: 0.6
  };

  return {
    solve: getAvg([noPause, noCollab, otherSolve], mode),
    apply: getAvg([noPause, noCollab, otherApply], mode),
  };
}

function getRange(num1, num2) {
  if (num1 < num2) {
    return `${num1.toFixed(1)}% - ${num2.toFixed(1)}%`;
  } else {
    return `${num2.toFixed(1)}% - ${num1.toFixed(1)}%`;
  }
}

function calculate() {
  const misalignmentMin = calculateMisalignmentLikelihood('min');
  const societalWeightsMin = calculateSocietalWeights('min');
  const misalignmentMax = calculateMisalignmentLikelihood('max');
  const societalWeightsMax = calculateSocietalWeights('max');

  const notSolvedMin = misalignmentMin + (societalWeightsMin.solve / 100);
  const minSolved = (100 - notSolvedMin);
  const solvedNotAppliedOrMisuseMin = minSolved - (minSolved * (societalWeightsMin.apply / 100));

  const notSolvedMax = misalignmentMax + (societalWeightsMax.solve / 100);
  const maxSolved = (100 - notSolvedMax);
  const solvedNotAppliedOrMisuseMax = maxSolved - (maxSolved * (societalWeightsMax.apply / 100));

  const minSNAM = solvedNotAppliedOrMisuseMin < solvedNotAppliedOrMisuseMax ? solvedNotAppliedOrMisuseMin : solvedNotAppliedOrMisuseMax;
  const maxSNAM = solvedNotAppliedOrMisuseMin < solvedNotAppliedOrMisuseMax ? solvedNotAppliedOrMisuseMax : solvedNotAppliedOrMisuseMin;

  const minRes = (notSolvedMin + minSNAM);
  const maxRes = (notSolvedMax + maxSNAM);

  const uncertainty = 30;

  const range = `
  <p>Not solved range: ${getRange(notSolvedMin, notSolvedMax)}</p>
  <p>Solved but not applied or misused range: ${getRange(solvedNotAppliedOrMisuseMin, solvedNotAppliedOrMisuseMax)}</p>
  <p>Not solved, applied, or misused (total) range: ${getRange(minRes, maxRes)}</p>
  <p>Solved range: ${getRange(minSolved, maxSolved)}</p>
  <p>Uncertainty on all values: ${uncertainty}%</p>`;
  document.querySelector('.range').innerHTML = range;
}

let customVars = [];

function addVariableElToDOM (name, min, max, weight) {
  const customVarsContainer = document.querySelector('.vars');
  const container = document.createElement('div');
  container.classList.add('customVar');
  container.id = name;

  const removeCustomVar = () => {
    customVars = customVars.filter(el => el.name !== name);
    document.querySelector(`#${name}`)?.remove();
  };

  const nameEl = document.createElement('p');
  nameEl.innerHTML = `Name: ${name}`;
  const minEl = document.createElement('p');
  minEl.innerHTML = `Min: ${min}`;;
  const maxEl = document.createElement('p');
  maxEl.innerHTML = `Max: ${max}`;;
  const weightEl = document.createElement('p');
  weightEl.innerHTML = `Weight: ${weight}`;
  const remove = document.createElement('div');
  remove.classList.add('removeEl');
  remove.onclick = removeCustomVar.bind(this);
  remove.innerHTML = '[X]';
  container.append(remove);
  container.append(nameEl);
  container.append(minEl);
  container.append(maxEl);
  container.append(weightEl);

  customVarsContainer.append(container);
}

function addVariable() {
  const nameEl = document.querySelector('#name');
  const nameVal = nameEl.value;
  nameEl.value = '';

  const minEl = document.querySelector('#min');
  const minVal = parseInt(minEl.value, 10);
  minEl.value = 0;

  const maxEl = document.querySelector('#max');
  const maxVal = parseInt(maxEl.value, 10);
  maxEl.value = 0;

  const weightEl = document.querySelector('#weight');
  const weightVal = parseFloat(weightEl.value);
  weightEl.value = 1;

  customVars.push({
    name: nameVal,
    min: minVal,
    max: maxVal,
    weight: weightVal
  });

  addVariableElToDOM(nameVal, minVal, maxVal, weightVal);
}

function calculateCustom() {
  console.log('customVars', customVars);
  const misalignmentMin = getAvg(customVars, 'min');
  const misalignmentMax = getAvg(customVars, 'max');

  const range = `Range: ${getRange(misalignmentMin, misalignmentMax)}`;
  console.log(range);
  document.querySelector('.range').innerHTML = range;
}

function saveCustomToCookies() {
  localStorage.setItem('customVars', JSON.stringify(customVars));
}

function loadCustomVars() {
  const loaded = JSON.parse(localStorage.getItem('customVars'));
  customVars = loaded;
  loaded.forEach(loadedVar => {
    addVariableElToDOM(loadedVar.name, loadedVar.min, loadedVar.max, loadedVar.weight);
  });
}
